java/integTest/src/test/java/glide/TestUtilities.java

```diff
@@ -76,4 +81,97 @@ public static Map<String, String> parseInfoResponseToMap(String serverInfo) {
         return RedisClusterClientConfiguration.builder()
                 .address(NodeAddress.builder().port(CLUSTER_PORTS[0]).build());
     }
+
+    /**
+     * Deep traverse and compare two objects, including comparing content of all nested collections
+     * recursively.
+     *
+     * @apiNote Maps comparison ignores their order, regardless of `orderMatters` argument. Map
+     *     entries could be reordered, but values stored in them compared according to this parameter.
+     */
+    public static void assertDeepEquals(Object expected, Object actual, boolean orderMatters) {
+        if (expected == null || actual == null) {
+            assertEquals(expected, actual);
+        } else if (expected.getClass().isArray()) {
+            var expectedArray = (Object[]) expected;
+            var actualArray = (Object[]) actual;
+            assertEquals(expectedArray.length, actualArray.length);
+            for (int i = 0; i < expectedArray.length; i++) {
+                assertDeepEquals(expectedArray[i], actualArray[i], orderMatters);
+            }
+        } else if (expected instanceof List) {
+            var expectedList = (List<?>) expected;
+            var actualList = (List<?>) actual;
+            assertEquals(expectedList.size(), actualList.size());
+            if (orderMatters) {
+                for (int i = 0; i < expectedList.size(); i++) {
+                    assertDeepEquals(expectedList.get(i), actualList.get(i), orderMatters);
+                }
+            } else {
+                assertTrue(expectedList.containsAll(actualList) && actualList.containsAll(expectedList));
+            }
+        } else if (expected instanceof Set) {
+            var expectedSet = (Set<?>) expected;
+            var actualSet = (Set<?>) actual;
+            var expectedArray = expectedSet.toArray();
+            var actualArray = actualSet.toArray();
+            assertEquals(expectedArray.length, actualArray.length);
+            if (orderMatters) {
+                for (int i = 0; i < expectedArray.length; i++) {
+                    assertDeepEquals(expectedArray[i], actualArray[i], orderMatters);
+                }
+            } else {
+                assertTrue(expectedSet.containsAll(actualSet) && actualSet.containsAll(expectedSet));
+            }
+        } else if (expected instanceof Map) {
+            var expectedMap = (Map<?, ?>) expected;
+            var actualMap = (Map<?, ?>) actual;
+            assertEquals(expectedMap.size(), actualMap.size());
+            assertDeepEquals(expectedMap.keySet(), actualMap.keySet(), false);
+            for (var key : expectedMap.keySet()) {
+                assertDeepEquals(expectedMap.get(key), actualMap.get(key), orderMatters);
+            }
+        } else {
+            assertEquals(expected, actual);
+        }
```

Comment:
reminder to double-check the test results with the changes in the transaction IT pr to be safe?